ocp4crd/oauth/oauth.go

```diff
@@ -88,48 +88,64 @@ func Generate(masterconfig configv1.MasterConfig) error {
 	for _, p := range auth.IdentityProviders {
 		p.Provider.Object, _, err = serializer.Decode(p.Provider.Raw, nil, nil)
 
+		if err != nil {
+			return nil, err
+		}
+
 		switch kind := p.Provider.Object.GetObjectKind().GroupVersionKind().Kind; kind {
 		case "HTPasswdPasswordIdentityProvider":
-			var idP identityProviderHTPasswd
-			var htpasswd configv1.HTPasswdPasswordIdentityProvider
-			_, _, _ = serializer.Decode(p.Provider.Raw, nil, &htpasswd)
-
-			idP.Type = "HTPasswd"
-			idP.Challenge = p.UseAsChallenger
-			idP.Login = p.UseAsLogin
-			idP.MappingMethod = p.MappingMethod
-			idP.HTPasswd.FileData.Name = htpasswd.File
-
+			idP := buildHTPasswdIP(serializer, p)
```

Comment:
This change has nothing to do with the commit implementing unit test, therefore it must be in separate commit.